homeassistant/components/roborock/coordinator.py

```diff
@@ -196,10 +206,44 @@ async def _async_setup(self) -> None:
                 flag=roborock_map.mapFlag,
                 name=roborock_map.name or f"Map {roborock_map.mapFlag}",
                 rooms={},
+                map_data=None,
+                image=None,
+                last_updated=dt_util.utcnow() - timedelta(seconds=IMAGE_CACHE_INTERVAL),
             )
             for roborock_map in (maps.map_info if (maps and maps.map_info) else ())
         }
 
+    async def update_map(self, bump_time: bool = True) -> None:
+        """Update the currently selected map."""
+        # The current map was set in the props update, so these can be done without
+        # worry of applying them to the wrong map.
+        if self.current_map is None:
+            # This exists as a safeguard/ to keep mypy happy.
+            return
+        response = await asyncio.gather(
+            *(
+                self.cloud_api.get_map_v1(),
+                self.set_current_map_rooms(),
+            ),
+            return_exceptions=True,
+        )
+        if not isinstance(response[0], bytes):
+            _LOGGER.debug("Failed to parse map contents: %s", response[0])
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="map_failure",
+            )
+        parsed_image, parsed_map = await self.parse_image(response[0])
+        if parsed_map is None or parsed_image is None:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="map_failure",
+            )
+        self.maps[self.current_map].map_data = parsed_map
```

Comment:
Pull out a local variable RoborockMapInfo then update the 3 variables to avoid accessing the map again 3 times.